src/models/connectionStore.ts

```diff
@@ -126,13 +126,16 @@ export class ConnectionStore {
     public addSavedPassword(credentialsItem: IConnectionCredentialsQuickPickItem): Promise<IConnectionCredentialsQuickPickItem> {
         let self = this;
         return new Promise<IConnectionCredentialsQuickPickItem>( (resolve, reject) => {
+            let isEmptyPasswordInput: boolean = typeof(credentialsItem.connectionCreds['emptyPasswordInput'] !== 'undefined')
```

Comment:
I think your parentheses are in the wrong place here. Also you can simplify checking for undefined by just using `credentialsItem.connectionCreds['emptyPasswordInput']` as if it were a boolean in the if statement